src/features/contactverzoek/ContactverzoekFormulier.vue

```diff
@@ -99,41 +97,80 @@
     <form-fieldset>
       <service-data-wrapper :data="vragenSets" class="container">
         <template #success="{ data }">
-          <template v-if="data && data.length > 0">
-            <label class="utrecht-form-label">
-              <span> Onderwerp </span>
-              <select
-                class="utrecht-select utrecht-select--html-select"
-                name="VragenSets"
-                v-model="form.vragenSetId"
-                @input="setActive"
+          <!-- Dropdown for selecting Onderwerp -->
+          <label class="utrecht-form-label">
+            <span>Onderwerp</span>
+            <select
+              class="utrecht-select utrecht-select--html-select"
+              name="VragenSets"
+              v-model="form.vragenSetId"
+              @change="setOnderwerp"
+            >
+              <option value="" selected>Geen</option>
+              <option
+                v-for="item in [...data].sort((a, b) => b.id - a.id)"
+                :key="item.id"
+                :value="item.id"
               >
-                <option v-for="item in data" :key="item.id" :value="item.id">
-                  {{ item.naam }}
-                </option>
-              </select>
-            </label>
-            <template v-if="form.contactVerzoekVragenSet">
-              <template
-                v-for="(item, index) in form.contactVerzoekVragenSet
-                  .vraagAntwoord"
-                :key="index"
-              >
-                <label class="utrecht-form-label">
-                  <span> {{ item.vraag }} </span>
-                  <input
-                    class="utrecht-textbox utrecht-textbox--html-input"
-                    type="text"
-                    v-model="item.antwoord"
-                    @input="setActive"
-                  />
-                </label>
-              </template>
+                {{ item.titel }}
+              </option>
+            </select>
+          </label>
+
+          <!-- Dynamic fields based on selected Onderwerp -->
+          <template v-if="form.contactVerzoekVragenSet">
+            <template
+              v-for="(item, index) in form.contactVerzoekVragenSet
+                .vraagAntwoord"
+              :key="index"
+            >
+              <label class="utrecht-form-label">
+                <span>{{ item.description }}</span>
+                <input
+                  v-if="isInputVraag(item)"
+                  class="utrecht-textbox utrecht-textbox--html-input"
+                  type="text"
+                  v-model="item.input"
```

Comment:
Dit hoef je niet aan te passen. het is meer een vraag: is dit niet onnodig complex? wat als ze allemaal dezelfde property item.value hebben dan kan je bij het uitlezen van de waardes gewoon alle values nemen. Alleen bij een checkboxes vraag hoef je iets anders te doen bij de verwerking.

Reply:
hier hebben we het gister over gehad :)